File: src/test/java/net/slipp/service/qna/FacebookServiceTest.java
package net.slipp.service.qna;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

import org.junit.Test;

public class FacebookServiceTest {

    @Test
    public void createLink_to_question() {
        final String url = "http://localhost:8080";
        FacebookService dut = new FacebookService() {
            @Override
            protected String createApplicationUrl() {
                return url;
            }
            
        };
        
        assertThat(dut.createLink(1L), is(url + "/questions/1"));
    }

    
    @Test
    public void createLink_to_answer() {
        final String url = "http://localhost:8080";
        FacebookService dut = new FacebookService() {
            @Override
            protected String createApplicationUrl() {
                return url;
            }
            
        };
        
        assertThat(dut.createLink(1L, 2L), is(url + "/questions/1#answer-2"));
    }
}

File: src/test/scala/net/slipp/service/qna/FacebookServiceTest.scala
package net.slipp.service.qna

import org.hamcrest.CoreMatchers.is
import org.junit.Assert.assertThat
import org.junit.Test

class FacebookServiceTest {
  @Test def createLink_to_question {
    val url: String = "http://localhost:8080"
    val dut: FacebookService = new FacebookService() {
      protected override def createApplicationUrl: String = {
        return url
      }
    }
    assertThat(dut.createLink(1L), is(url + "/questions/1"))
  }

  @Test def createLink_to_answer {
    val url: String = "http://localhost:8080"
    val dut: FacebookService = new FacebookService() {
      protected override def createApplicationUrl: String = {
        return url
      }
    }
    assertThat(dut.createLink(1L, 2L), is(url + "/questions/1#answer-2"))
  }
}
